aepsych/models/utils.py

```diff
@@ -61,72 +65,85 @@ def compute_p_quantile(
 
 def select_inducing_points(
```

Comment:
Consider carefully if this whole utility function is needed.  A lot of the logic here should probably be placed in AutoAllocator instead which can create new allocators within itself to manage auto selection.